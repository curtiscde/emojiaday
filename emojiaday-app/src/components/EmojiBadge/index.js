import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import Badge from '@material-ui/core/Badge';
import PropTypes from 'prop-types';
import { Emoji } from 'emoji-mart';

const EmojiBadge = (props) => {
  const styles = {
    badge: {
      top: 1,
      right: -15,
      border: '2px solid #ccc',
    },
  };

  const getBackgroundColor = (index) => {
    switch (index) {
      case 0: {
        return 'rgba(255, 215, 0, 0.3)';
      }
      case 1: {
        return 'rgba(211, 211, 211, 0.3)';
      }
      case 2: {
        return 'rgba(205, 127, 50, 0.3)';
      }
      default: {
        return null;
      }
    }
  };

  if (!props.emoji) return null;

  return (
    <IconButton key={props.emoji._id} style={{ backgroundColor: getBackgroundColor(props.index) }}>
      {
        props.emoji.count > 10
          ? (
            <Badge badgeContent={props.emoji.count} color="primary" classes={{ badge: styles.badge }}>
              <Emoji emoji={props.emoji._id} set="twitter" size={32} />
            </Badge>
          )
          : (
            <Emoji emoji={props.emoji._id} set="twitter" size={32} />
          )
      }
    </IconButton>
  );
};

EmojiBadge.propTypes = {
  emoji: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
};

export default EmojiBadge;
