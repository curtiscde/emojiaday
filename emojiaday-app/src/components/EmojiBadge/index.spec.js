import React from 'react';
import renderer from 'react-test-renderer';
import EmojiBadge from './index';

describe('EmojiBadge', () => {
  it('matches snapshot when empty', () => {
    const tree = renderer
      .create(<EmojiBadge />);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when emoji passed', () => {
    const emoji = {
      _id: 'smile',
    };

    const tree = renderer
      .create(<EmojiBadge emoji={emoji} />);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when emoji passed with count', () => {
    const emoji = {
      _id: 'smile',
      count: 5,
    };

    const tree = renderer
      .create(<EmojiBadge emoji={emoji} />);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when emoji passed with count greater than 10', () => {
    const emoji = {
      _id: 'smile',
      count: 15,
    };

    const tree = renderer
      .create(<EmojiBadge emoji={emoji} />);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when emoji and index of 0 passed', () => {
    const emoji = {
      _id: 'smile',
    };

    const tree = renderer
      .create(<EmojiBadge emoji={emoji} index={0} />);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when emoji and index of 1 passed', () => {
    const emoji = {
      _id: 'smile',
    };

    const tree = renderer
      .create(<EmojiBadge emoji={emoji} index={1} />);
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot when emoji and index of 2 passed', () => {
    const emoji = {
      _id: 'smile',
    };

    const tree = renderer
      .create(<EmojiBadge emoji={emoji} index={2} />);
    expect(tree).toMatchSnapshot();
  });
});
