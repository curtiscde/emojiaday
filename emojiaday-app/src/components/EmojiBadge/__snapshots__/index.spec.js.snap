// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojiBadge matches snapshot when emoji and index of 0 passed 1`] = `
<button
  className="MuiButtonBase-root-7 MuiIconButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 215, 0, 0.3)",
    }
  }
  tabIndex="0"
  type="button"
>
  <span
    className="MuiIconButton-label-6"
  >
    <span
      className="emoji-mart-emoji"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      title={null}
    >
      <span
        style={
          Object {
            "backgroundImage": "url(https://unpkg.com/emoji-datasource-twitter@4.0.4/img/twitter/sheets-256/64.png)",
            "backgroundPosition": "58.8235294117647% 54.90196078431372%",
            "backgroundSize": "5200%",
            "display": "inline-block",
            "height": 32,
            "width": 32,
          }
        }
      />
    </span>
  </span>
  <span
    className="MuiTouchRipple-root-10"
  />
</button>
`;

exports[`EmojiBadge matches snapshot when emoji and index of 1 passed 1`] = `
<button
  className="MuiButtonBase-root-7 MuiIconButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(211, 211, 211, 0.3)",
    }
  }
  tabIndex="0"
  type="button"
>
  <span
    className="MuiIconButton-label-6"
  >
    <span
      className="emoji-mart-emoji"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      title={null}
    >
      <span
        style={
          Object {
            "backgroundImage": "url(https://unpkg.com/emoji-datasource-twitter@4.0.4/img/twitter/sheets-256/64.png)",
            "backgroundPosition": "58.8235294117647% 54.90196078431372%",
            "backgroundSize": "5200%",
            "display": "inline-block",
            "height": 32,
            "width": 32,
          }
        }
      />
    </span>
  </span>
  <span
    className="MuiTouchRipple-root-10"
  />
</button>
`;

exports[`EmojiBadge matches snapshot when emoji and index of 2 passed 1`] = `
<button
  className="MuiButtonBase-root-7 MuiIconButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(205, 127, 50, 0.3)",
    }
  }
  tabIndex="0"
  type="button"
>
  <span
    className="MuiIconButton-label-6"
  >
    <span
      className="emoji-mart-emoji"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      title={null}
    >
      <span
        style={
          Object {
            "backgroundImage": "url(https://unpkg.com/emoji-datasource-twitter@4.0.4/img/twitter/sheets-256/64.png)",
            "backgroundPosition": "58.8235294117647% 54.90196078431372%",
            "backgroundSize": "5200%",
            "display": "inline-block",
            "height": 32,
            "width": 32,
          }
        }
      />
    </span>
  </span>
  <span
    className="MuiTouchRipple-root-10"
  />
</button>
`;

exports[`EmojiBadge matches snapshot when emoji passed 1`] = `
<button
  className="MuiButtonBase-root-7 MuiIconButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": null,
    }
  }
  tabIndex="0"
  type="button"
>
  <span
    className="MuiIconButton-label-6"
  >
    <span
      className="emoji-mart-emoji"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      title={null}
    >
      <span
        style={
          Object {
            "backgroundImage": "url(https://unpkg.com/emoji-datasource-twitter@4.0.4/img/twitter/sheets-256/64.png)",
            "backgroundPosition": "58.8235294117647% 54.90196078431372%",
            "backgroundSize": "5200%",
            "display": "inline-block",
            "height": 32,
            "width": 32,
          }
        }
      />
    </span>
  </span>
  <span
    className="MuiTouchRipple-root-10"
  />
</button>
`;

exports[`EmojiBadge matches snapshot when emoji passed with count 1`] = `
<button
  className="MuiButtonBase-root-7 MuiIconButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": null,
    }
  }
  tabIndex="0"
  type="button"
>
  <span
    className="MuiIconButton-label-6"
  >
    <span
      className="emoji-mart-emoji"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      title={null}
    >
      <span
        style={
          Object {
            "backgroundImage": "url(https://unpkg.com/emoji-datasource-twitter@4.0.4/img/twitter/sheets-256/64.png)",
            "backgroundPosition": "58.8235294117647% 54.90196078431372%",
            "backgroundSize": "5200%",
            "display": "inline-block",
            "height": 32,
            "width": 32,
          }
        }
      />
    </span>
  </span>
  <span
    className="MuiTouchRipple-root-10"
  />
</button>
`;

exports[`EmojiBadge matches snapshot when emoji passed with count greater than 10 1`] = `
<button
  className="MuiButtonBase-root-7 MuiIconButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "backgroundColor": null,
    }
  }
  tabIndex="0"
  type="button"
>
  <span
    className="MuiIconButton-label-6"
  >
    <span
      className="MuiBadge-root-17"
    >
      <span
        className="emoji-mart-emoji"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        title={null}
      >
        <span
          style={
            Object {
              "backgroundImage": "url(https://unpkg.com/emoji-datasource-twitter@4.0.4/img/twitter/sheets-256/64.png)",
              "backgroundPosition": "58.8235294117647% 54.90196078431372%",
              "backgroundSize": "5200%",
              "display": "inline-block",
              "height": 32,
              "width": 32,
            }
          }
        />
      </span>
      <span
        className="MuiBadge-badge-18 [object Object] MuiBadge-colorPrimary-19"
      >
        15
      </span>
    </span>
  </span>
  <span
    className="MuiTouchRipple-root-10"
  />
</button>
`;

exports[`EmojiBadge matches snapshot when empty 1`] = `null`;
